### PR TITLE
Fix half of emscripten's failing tests

### PR DESCRIPTION
**Pull request by tomaka**
Before this PR:

> test result: FAILED. 2039 passed; 327 failed; 2 ignored; 0 measured

After:

> test result: FAILED. 2232 passed; 134 failed; 2 ignored; 0 measured

r? @brson

### CODE REVIEW

File: src/compiletest/runtest.rs [hunk 1]
```diff
@@ -1229,11 +1229,13 @@ fn compose_and_run_compiler(config: &Config, props: &TestProps,
...
             //
-            // For targets like MUSL, however, there is no support for dynamic
-            // libraries so we just go back to building a normal library. Note,
-            // however, that if the library is built with `force_host` then it's
-            // ok to be a dylib as the host should always support dylibs.
-            if config.target.contains("musl") && !aux_props.force_host {
+            // For targets like MUSL or Emscripten, however, there is no support for
+            // dynamic libraries so we just go back to building a normal library. Note,
+            // however, that for MUSL if the library is built with `force_host` then
+            // it's ok to be a dylib as the host should always support dylibs.
```
Comment (oli-obk) [inline src/compiletest/runtest.rs]: why isn't the `force_host` statement also true for emscripten?

Comment (tomaka) [inline src/compiletest/runtest.rs]: The comment says "if the library is built with `force_host` then it's ok to be a dylib as the host should always support dylibs", but emscripten doesn't support dynamic libraries at all.

Comment (oli-obk) [inline src/compiletest/runtest.rs]: yes, but the host can never be emscripten, right? That would mean we'd have a rustc that runs on emscripten.

I think `force_host` is just for rustc's unit tests when compiling a compiler plugin.

Comment (tomaka) [inline src/compiletest/runtest.rs]: > That would mean we'd have a rustc that runs on emscripten.

Well, why not? :pig:

**Pull request merged**